app/static/ringing_room.js

```diff
@@ -1445,41 +1456,57 @@ $(document).ready(function () {
                 this.method_name = "";
             },
 
-            update_comp_suggestions: function (partial_comp_name) {
-                if (partial_comp_name == "") {
-                    this.current_complib_comp = undefined;
-                    this.complib_error = "Start typing a comp ID...";
+            update_comp_suggestions: function (url_box_contents) {
+                // If the user changed the `Composition` box, then we always invalidate the current
+                // Wheatley comp until CompLib replies to our HTTP request.  This should almost
+                // certainly be so fast that the user doesn't see it.
+                this.current_complib_comp = undefined;
+                // If the box is empty, then clear the error box
+                if (url_box_contents == "") {
+                    this.complib_message = "";
+                    return;
+                }
+                // Normalise the URL by removing `https://` or `http://` off the front (sometimes
+                // it's missed off when you copy URLs).  We'll add `https://` again before making
+                // HTTP requests
+                const complib_url_without_http = url_box_contents
+                    .replace("https://", "")
+                    .replace("http://", "")
+                    .toLowerCase();
+                const url_segments = complib_url_without_http.split("/");
+                // Do some basic validation of the URL
+                if (!complib_url_without_http.startsWith("complib.org")) {
```

Comment:
Should the input box continue to accept just an id for backwards compatibility / familiarity?